### PR TITLE
Fix dangling footstep bug, a couple misc bugs

### PR DESCRIPTION
**Pull request by evanw555**
Fixes the issue where duplicate footsteps weren't getting deleted
when the player and all its related points should've been removed
from the map. Also fixes unrelated NPE and CME bugs.

### CODE REVIEW

**Review by davidvorona — APPROVED** [review]

**Comment by ReinierAarnink1996** [comment]
Hey man is this footstepd thing live yet? Just checking in to see if there is a next version with more features..
My gim team loves the plugin, they enjoy checking out who is where, as do I lol. One wont use it, he doesnt like to be located lol

**Comment by davidvorona** [comment]
Still waiting on approval from the Runelite team...once they do that, this will be live. It's really dope.

Your friend should enable ghost mode ;)

**Comment by evanw555** [comment]
@ReinierAarnink1996 We're waiting on the Runelite team to approve the rollout of v2.0 to plugin-hub: https://github.com/runelite/plugin-hub/pull/2314

Perhaps you can comment to +1 it, but I'm not sure if that would affect how fast they'll get to it, as they seem to review these sort of PRs in batches.

**Pull request merged**
